### ISSUE TITLE
Data Race in SCTP

### ISSUE THREAD

**Issue by maxhawkins** [pions/webrtc#254]
Found this data race in sctp while running tests on Travis.

```
==================
WARNING: DATA RACE
Read at 0x00c0002f7810 by goroutine 110:
  github.com/pions/webrtc/internal/sctp.(*Association).handleChunk()
      /home/travis/gopath/src/github.com/pions/webrtc/internal/sctp/association.go:520 +0x1579
  github.com/pions/webrtc/internal/sctp.(*Association).HandleInbound()
      /home/travis/gopath/src/github.com/pions/webrtc/internal/sctp/association.go:128 +0x1df
  github.com/pions/webrtc/internal/network.(*port).handleSCTP()
      /home/travis/gopath/src/github.com/pions/webrtc/internal/network/port-receive.go:134 +0x138
  github.com/pions/webrtc/internal/network.(*port).handleDTLS()
      /home/travis/gopath/src/github.com/pions/webrtc/internal/network/port-receive.go:147 +0xb8b
  github.com/pions/webrtc/internal/network.(*port).networkLoop()
      /home/travis/gopath/src/github.com/pions/webrtc/internal/network/port-receive.go:216 +0x592
Previous write at 0x00c0002f7810 by goroutine 29:
  github.com/pions/webrtc/internal/sctp.(*Association).setState()
      /home/travis/gopath/src/github.com/pions/webrtc/internal/sctp/association.go:284 +0xa6
  github.com/pions/webrtc/internal/sctp.(*Association).Connect()
      /home/travis/gopath/src/github.com/pions/webrtc/internal/sctp/association.go:298 +0xe6
  github.com/pions/webrtc/internal/network.(*Manager).handleDTLSState()
      /home/travis/gopath/src/github.com/pions/webrtc/internal/network/manager.go:125 +0xa2
  github.com/pions/webrtc/internal/network.(*Manager).handleDTLSState-fm()
      /home/travis/gopath/src/github.com/pions/webrtc/internal/network/manager.go:76 +0x4a
Goroutine 110 (running) created at:
  github.com/pions/webrtc/internal/network.newPort()
      /home/travis/gopath/src/github.com/pions/webrtc/internal/network/port.go:38 +0x232
  github.com/pions/webrtc/internal/network.NewManager()
      /home/travis/gopath/src/github.com/pions/webrtc/internal/network/manager.go:85 +0x5a9
  github.com/pions/webrtc.New()
      /home/travis/gopath/src/github.com/pions/webrtc/rtcpeerconnection.go:147 +0x380
  github.com/pions/webrtc.TestCreateOfferAnswer()
      /home/travis/gopath/src/github.com/pions/webrtc/rtcpeerconnection_test.go:312 +0x87
  testing.tRunner()
      /home/travis/.gimme/versions/go1.11.2.linux.amd64/src/testing/testing.go:827 +0x162
Goroutine 29 (finished) created at:
  github.com/pions/webrtc/internal/dtls.(*State).setState()
      /home/travis/gopath/src/github.com/pions/webrtc/internal/dtls/dtls.go:117 +0x11a
  github.com/pions/webrtc/internal/dtls.(*State).HandleDTLSPacket()
      /home/travis/gopath/src/github.com/pions/webrtc/internal/dtls/dtls.go:181 +0x395
  github.com/pions/webrtc/internal/network.(*port).handleDTLS()
      /home/travis/gopath/src/github.com/pions/webrtc/internal/network/port-receive.go:140 +0x150
  github.com/pions/webrtc/internal/network.(*port).networkLoop()
      /home/travis/gopath/src/github.com/pions/webrtc/internal/network/port-receive.go:216 +0x592
==================
Found 1 data race(s)
FAIL	github.com/pions/webrtc	5.157s
```

**Comment by Sean-Der** [comment]
@kc5nra mind checking this out?

**Comment by backkem** [comment]
Reopening since the change was reverted.

**Comment by backkem** [comment]
This should be resolved by #267. Feel free to open a new issue when finding more races.
